src/internal/utils.js

```diff
@@ -48,7 +48,12 @@ export function validate_store(store, name) {
 }
 
 export function subscribe(component, store, callback) {
-	component.$$.on_destroy.push(store.subscribe(callback));
+	let unsub = store.subscribe(callback);
+	// Prevent memory leaks for RxJS users.
```

Comment:
I was initially not supportive of this PR, but it seems to go beyond supporting just RxJS and it would be ideal if this comment would reflect that, so as not to give others the wrong impression that Svelte specifically supports RxJS rather than `Observable`s (and other things with `unsubscribe`) in general.